### ISSUE TITLE
`/orders`(or `/orders/{id}`) endpoint missing "state" and "status" fields

### ISSUE THREAD

**Issue by Naberal** [magento/magento2#37807]
### Preconditions and environment

- Magento version:  I reproduced on 2.4.6-p1
- It caused by broken DB data 

### Steps to reproduce

I reproduce in this way:

1. create new order
2. in DB in table `sales_order` set "status" and "state" as null
3. make a request in postman using the `/orders`(or `/orders/{id}`).

### Expected result

Per the [rest api docs](https://adobe-commerce.redoc.ly/2.4.6-admin/tag/orders#operation/GetV1Orders) there should be "state" and "status" fields on the response data and their type is a string, in such case empty string.

### Actual result

"state" and "status" fields missing entirely.

### Additional information

with such broken DB data, Magento works Ok, but for those who use these endpoints, it is unexpected behavior.

### Triage and priority

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [X] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

**Comment by engcom-Bravo** [comment]
Hi @Naberal,

Thank you for reporting and collaboration.

Verified the issue on Magento 2.4-develop instance and the issue is reproducible.Kindly refer the screenshots.

**Steps to reproduce**

- create an order as a customer from UI make note of order id  login to data base and check the table 'sales_order' and set status and state' field as null
- login to admin user using api request and note the access token received 
- Execute the api query orders/{id} to check the order status.



<img width="1131" alt="Screenshot 2023-07-31 at 8 46 20 PM" src="https://github.com/magento/magento2/assets/51680745/b10ed044-2dec-4712-9be7-71798e6ac9e9">


previously it was there before changing the their status in database

<img width="1119" alt="Screenshot 2023-07-31 at 8 45 29 PM" src="https://github.com/magento/magento2/assets/51680745/9d23a39d-ab3a-4720-9458-410731b9fe29">



After updating in database "state" and "status" fields missing entirely.

Hence Confirming the issue.

Thanks.

**Comment by Aditya-Prakash-Talluru** [comment]
@magento I am working on this

**Comment by Naberal** [comment]
@Aditya-Prakash-Talluru 
I don't understand. What do you mean?

**Comment by Aditya-Prakash-Talluru** [comment]
@Naberal 

I mean when the value is updated to null it means that it doesn’t have any value. In this scenario it is better to ignore the label display in the api result. Incase of an empty string the label is visible with value as empty string.

**Comment by Naberal** [comment]
> In this scenario it is better to ignore the label display in the api result.

no, return `null` and correct [rest api docs](https://adobe-commerce.redoc.ly/2.4.6-admin/tag/orders#operation/GetV1Orders), that some fields in response may be `null` and let the person who did request decide what to do with it.

**Comment by engcom-Delta** [comment]
Hi @Naberal ,

Thanks for your reporting and collaboration.
We have re-verified the issue in latest 2.4-develop instance hence reconfirming the issue.
Kindly refer the screenshot.

Steps to reproduce

1. create an order as a customer from UI make note of order id login to data base and check the table 'sales_order' and set status and state' field as null
Before

<img width="154" height="94" alt="Image" src="https://github.com/user-attachments/assets/c721f51d-67ae-4b70-97dd-871068e63752" />

<img width="533" height="397" alt="Image" src="https://github.com/user-attachments/assets/54d6e63e-842c-4f9e-9e33-006e0662ed45" />

Set field as null

<img width="317" height="99" alt="Image" src="https://github.com/user-attachments/assets/1afe390e-885b-4397-8af6-d573d88ce73e" />

2. login to admin user using api request and note the access token received
3. Execute the api query orders/{id} to check the order status.

<img width="646" height="335" alt="Image" src="https://github.com/user-attachments/assets/aa8c5a3c-62d2-4386-ac02-33e4dac569e7" />

Hence reconfirming the issue.
Thanks
